src/app/prt-staking/provider.tsx

```diff
@@ -96,68 +131,103 @@ export const PrtStakingContextProvider = ({ children, token }: Props) => {
   useEffect(() => {
     async function fetchTokenData() {
       const [tvl, cumulativeRevenue] = await Promise.all([
-        fetchTvl(token.tokenData.symbol),
-        fetchCumulativeRevenue(token.tokenData.address),
+        fetchTvl(token.rewardTokenData.symbol),
+        fetchCumulativeRevenue(token.rewardTokenData.address),
       ])
       setTvl(tvl)
       setCumulativeRevenue(cumulativeRevenue)
     }
     fetchTokenData()
-  }, [token.tokenData.address, token.tokenData.symbol])
+  }, [token.rewardTokenData.address, token.rewardTokenData.symbol])
 
   const claimPrts = useCallback(async () => {
     if (!walletClient) return
     await walletClient.writeContract({
       abi: PrtStakingAbi,
-      address: prtTokenAddress, // rewardToken
+      address: stakedTokenAddress,
       functionName: 'claim',
     })
-  }, [prtTokenAddress, walletClient])
+  }, [stakedTokenAddress, walletClient])
 
   const stakePrts = useCallback(
     async (amount: bigint) => {
       if (!walletClient) return
-      await walletClient.writeContract({
-        abi: PrtStakingAbi,
-        address: prtTokenAddress,
-        functionName: 'stake',
-        args: [amount],
-      })
+      // FIXME: Need to add proper UX handling for case where staking is disabled
+      if (!canStake) return
+      if (isApprovedStaker) {
+        await walletClient.writeContract({
+          abi: PrtStakingAbi,
+          address: stakeTokenAddress,
+          functionName: 'stake',
+          args: [amount],
+        })
+      } else {
+        if (!stakeSignMessage) return
+        if (!stakeDomain) return
+        const signature = await walletClient.signTypedData({
+          // FIXME: Domain arg typing/format
+          domain: stakeDomain,
```

Comment:
Domain value returned from the contract doesn't match the expected type here, will need to investigate

Reply:
Usually this is a struct (https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator) with at least `version`and `verifyingContract` defined.

Reply:
As mentioned on discord, I would just hard-code this. No need to fetch every time.